### PR TITLE
Check if user is already a veto mediator before reselection

### PR DESCRIPTION
**Pull request by venix12**

### CODE REVIEW

File: routes/vetoes.js [hunk 1]
```diff
@@ -181,26 +181,27 @@ router.post('/replaceMediator/:id', api.isNat, api.isNotSpectator, async (req, r
     let v = await vetoesService.query({ _id: req.params.id }, defaultPopulate);
     let currentMediators = [];
     v.mediations.forEach(mediation => {
-        if(mediation.id != req.body.mediationId){
```
Comment (pishifat) [inline routes/vetoes.js]: you can still include the current mediation since you dont want to replace with the same user

File: routes/vetoes.js [hunk 2]
```diff
@@ -181,26 +181,27 @@ router.post('/replaceMediator/:id', api.isNat, api.isNotSpectator, async (req, r
...
 
     const allUsers = await usersService.getAllMediators();
     for (let i = 0; i < allUsers.length; i++) {
         let user = allUsers[i];
         if (
-            user.modes.indexOf(v.mode) >= 0 &&
-            user.probation.indexOf(v.mode) < 0 &&
-            user.osuId != v.vetoer.osuId &&
-            user.osuId != v.beatmapMapperId
+            !currentMediators.includes(String(user._id))
```
Comment (pishifat) [inline routes/vetoes.js]: use user.id, don't need to stringify the _id

File: routes/vetoes.js (hunk 1 above)
Comment (venix12) [reply routes/vetoes.js]: ye that's literally what it does after removing the ``if`` 🤔

File: routes/vetoes.js (hunk 2 above)
Comment (venix12) [reply routes/vetoes.js]: seems like it's returning ``undefined`` though

Comment (pishifat) [reply routes/vetoes.js]: works fine for me? what exactly are you doing hmmm

Comment (venix12) [reply routes/vetoes.js]: > use user.id, don't need to stringify the _id

well seems like you mean ``_id`` lol

**Comment by pishifat** [comment]
nah i mean id

**Comment by venix12** [comment]
no idea, maybe im doing something wrong

**Pull request merged**